11/debian-9/rootfs/libpostgresql.sh

```diff
@@ -0,0 +1,694 @@
+  fi
+}
+
+########################
+# Load global variables used on PostgreSQL configuration.
+# Globals:
+#   POSTGRESQL_*
+# Arguments:
+#   None
+# Returns:
+#   Series of exports to be used as 'eval' arguments
+#########################
+postgresql_env() {
+    declare_env_alias() {
+      local -r alias="${1:?missing environment variable alias}"
+      local -r original="${2:?missing original environment variable}"
+
+      if env | grep -q "${original}"; then
+          cat << EOF 
+export $alias="${!original}"
+EOF
+      fi
+    }
+  
+    # Alias created for official PostgreSQL image compatibility
+    declare_env_alias POSTGRESQL_DATABASE POSTGRES_DB
+    declare_env_alias POSTGRESQL_USERNAME POSTGRES_USER
+    declare_env_alias POSTGRESQL_DATA_DIR PGDATA
+
+    local -r suffixes=(
+      "PASSWORD" "INITDB_WAL_DIR" "INITDB_ARGS" "CLUSTER_APP_NAME" 
+      "MASTER_HOST" "MASTER_PORT_NUMBER" "NUM_SYNCHRONOUS_REPLICAS" 
+      "PORT_NUMBER" "REPLICATION_MODE" "REPLICATION_PASSWORD" "REPLICATION_USER"
+      "SYNCHRONOUS_COMMIT_MODE" "PASSWORD_FILE" "REPLICATION_PASSWORD_FILE" "INIT_MAX_TIMEOUT"
+    )
+    for s in "${suffixes[@]}"; do
+      declare_env_alias "POSTGRESQL_${s}" "POSTGRES_${s}"
+    done
+
+    cat <<"EOF"
+# Paths
+export POSTGRESQL_VOLUME_DIR="/bitnami/postgresql"
+export POSTGRESQL_DATA_DIR="${POSTGRESQL_DATA_DIR:-$POSTGRESQL_VOLUME_DIR/data}"
+export POSTGRESQL_BASE_DIR="/opt/bitnami/postgresql"
+export POSTGRESQL_CONF_DIR="$POSTGRESQL_BASE_DIR/conf"
+export POSTGRESQL_MOUNTED_CONF_DIR="/bitnami/postgresql/conf"
+export POSTGRESQL_CONF_FILE="$POSTGRESQL_CONF_DIR/postgresql.conf"
+export POSTGRESQL_PGHBA_FILE="$POSTGRESQL_CONF_DIR/pg_hba.conf"
+export POSTGRESQL_RECOVERY_FILE="$POSTGRESQL_DATA_DIR/recovery.conf"
+export POSTGRESQL_LOG_DIR="$POSTGRESQL_BASE_DIR/logs"
+export POSTGRESQL_LOG_FILE="$POSTGRESQL_LOG_DIR/postgresql.log"
+export POSTGRESQL_TMP_DIR="$POSTGRESQL_BASE_DIR/tmp"
+export POSTGRESQL_PID_FILE="$POSTGRESQL_TMP_DIR/postgresql.pid"
+export POSTGRESQL_BIN_DIR="$POSTGRESQL_BASE_DIR/bin"
+export POSTGRESQL_INITSCRIPTS_DIR=/docker-entrypoint-initdb.d
+export PATH="$POSTGRESQL_BIN_DIR:$PATH"
+
+# Users
+export POSTGRESQL_DAEMON_USER="postgresql"
+export POSTGRESQL_DAEMON_GROUP="postgresql"
+
+# Settings
+export POSTGRESQL_INIT_MAX_TIMEOUT=${POSTGRESQL_INIT_MAX_TIMEOUT:-60}
+export POSTGRESQL_CLUSTER_APP_NAME=${POSTGRESQL_CLUSTER_APP_NAME:-walreceiver}
+export POSTGRESQL_DATABASE="${POSTGRESQL_DATABASE:-}"
+export POSTGRESQL_INITDB_ARGS="${POSTGRESQL_INITDB_ARGS:-}"
+export ALLOW_EMPTY_PASSWORD="${ALLOW_EMPTY_PASSWORD:-yes}"
+export POSTGRESQL_INITDB_WAL_DIR="${POSTGRESQL_INITDB_WAL_DIR:-}"
+export POSTGRESQL_MASTER_HOST="${POSTGRESQL_MASTER_HOST:-}"
+export POSTGRESQL_MASTER_PORT_NUMBER="${POSTGRESQL_MASTER_PORT_NUMBER:-5432}"
+export POSTGRESQL_NUM_SYNCHRONOUS_REPLICAS="${POSTGRESQL_NUM_SYNCHRONOUS_REPLICAS:-0}"
+export POSTGRESQL_PORT_NUMBER="${POSTGRESQL_PORT_NUMBER:-5432}"
```

Comment:
This variable is defined but unused. Should we allow changing Postgres port?

Reply:
If it was allowed before, then yes. I wouldn't allow it in the chart, though

Reply:
Previously it was not possible but we were exporting the env var, so it was confusing. So I would fix it one way or another. Allow changing Postgres port or don't expose the env var.